anvil/config.py

```diff
@@ -156,26 +157,47 @@ def produce_training_geometry(self, training_context):
             _, geometry = self.parse_from_(None, "geometry", write=False)
         return geometry
 
-    @explicit_node
-    def produce_loaded_optimizer(self, optimizer):
-        """Returns an action which itself returns a torch.optim.Optimizer object
-        that knows about the current state of the model."""
+    def parse_optimizer(self, optimizer):
         try:
-            return OPTIMIZER_OPTIONS[optimizer]
+            optim_class = getattr(torch.optim, optimizer)  # could make case-insensitive
         except KeyError:
+            raise ConfigError(f"Invalid optimizer {optimizer}. Consult torch.optim.")
+        return optim_class
+
+    def parse_optimizer_params(self, params, optimizer):
+        try:
+            test = optimizer(
+                [{"params": []}],
+                **params,
+            )
+        except TypeError as error:
+            print(error)
             raise ConfigError(
-                f"Invalid optimizer {optimizer}", optimizer, OPTIMIZER_OPTIONS.keys()
+                f"Invalid optimizer keyword argument dict. Consult documentation for {optimizer}."
             )
+        return params
```

Comment:
But this I think is pointless and I regret taking the time spent writing it! It would be much easier to just not attempt to catch every error in the parser, and just let pytorch throw it's errors during runtime if it needs to. So, what I mean is, I think we should use the parser exclusively to check for things that don't make sense for the *application* (e.g. negative lattice length), and don't try to catch every bad input. If you agree, I'll change these parse functions.

Reply:
I do agree!

Reply:
if you put typing in the signature then reportengine will check for that then just check for sensible values

Reply:
if you get rid of the checking you can also scrap optimizer from signature right?

Reply:
Yeah I'll get rid.